### PR TITLE
[Xamarin] Update xamarin project to be built

### PR DESCRIPTION
**Pull request by almedina-ms**
Updates the xamarin project to be able to build and run

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/AdaptiveCards/pull/3344)

### CODE REVIEW

File: source/ios/README.md [hunk 1]
```diff
@@ -1,5 +1,31 @@
 # Adaptive Cards in iOS
 
+<<<<<<< HEAD
```
Comment (nesalang) [inline source/ios/README.md]: bad merge here and else where

File: source/xamarin/AdaptiveCards.Rendering.Xamarin.iOS/Structs.cs [hunk 2]
```diff
@@ -0,0 +1,70 @@
+﻿using System;
+using ObjCRuntime;
+using Foundation;
+
+
+namespace AdaptiveCards.Rendering.Xamarin.iOS
```
Comment (nesalang) [inline source/xamarin/AdaptiveCards.Rendering.Xamarin.iOS/Structs.cs]: Nit: Why is the file named Structs.cs?

File: source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -136,7 +136,6 @@
 		F4071C7E1FCCBAEF00AF4FEA /* ActionParserRegistration.cpp in Sources */ = {isa = PBXBuildFile; fileRef = F4071C781FCCBAEF00AF4FEA /* ActionParserRegistration.cpp */; };
 		F4071C7F1FCCBAEF00AF4FEA /* ElementParserRegistration.cpp in Sources */ = {isa = PBXBuildFile; fileRef = F4071C791FCCBAEF00AF4FEA /* ElementParserRegistration.cpp */; };
 		F423C0BF1EE1FBAA00905679 /* AdaptiveCards.framework in Frameworks */ = {isa = PBXBuildFile; fileRef = F423C0B51EE1FBA900905679 /* AdaptiveCards.framework */; };
```
Comment (nesalang) [inline source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards.xcodeproj/project.pbxproj]: iOS is not building. AdaptiveCards.h should be part of public hearders list.

File: source/android/adaptivecards/build.gradle [hunk 4]
```diff
@@ -91,7 +91,7 @@ publishing {
             url 'https://microsoft.pkgs.visualstudio.com/_packaging/AdaptiveCards/maven/v1'
             credentials {
                 username "AZURE_ARTIFACTS"
-                password System.getenv("AZURE_ARTIFACTS_ENV_ACCESS_TOKEN") ?: "${azureArtifactsGradleAccessToken}"
+                // password System.getenv("AZURE_ARTIFACTS_ENV_ACCESS_TOKEN") ?: "${azureArtifactsGradleAccessToken}"
```
Comment (paulcam206) [inline source/android/adaptivecards/build.gradle]: > // password System.getenv("AZURE_ARTIFACTS_ENV_ACCESS_TOKEN") ?: "${azureArtifactsGradleAccessToken}" [](start = 15, length = 102)

is this needed?

File: source/ios/AdaptiveCards/ADCIOSVisualizer/ADCIOSVisualizerTests/ADCIOSVisualizerTests.mm [hunk 5]
```diff
@@ -7,7 +7,7 @@
 //
 
 #import <XCTest/XCTest.h>
-#import <AdaptiveCards/ACFramework.h>
+AdaptiveCards.h>/AdaptiveCards.h>
```
Comment (paulcam206) [inline source/ios/AdaptiveCards/ADCIOSVisualizer/ADCIOSVisualizerTests/ADCIOSVisualizerTests.mm]: this doesn't look right...

File: source/xamarin/AdaptiveCards.Rendering.Xamarin.iOS.Sample/README.md [hunk 6]
```diff
@@ -0,0 +1,10 @@
+# Adaptive Cards Xamarin iOS Sample #
+
+This folder contains the [Xamarin](https://xamarin.com) sample app for [Adaptive Cards](https://adaptivecards/io) for iOS. This sample uses the BotConnection for retrieving the payloads.
+
+### How do I fix this? ###
+
+#### Couldn't isntall app: file not found ####
```
Comment (almedina-ms) [inline source/xamarin/AdaptiveCards.Rendering.Xamarin.iOS.Sample/README.md]: typo in install

**Comment by shalinijoshi19** [comment]
Can we clean up the commit history a bit to aid in code reviews?

File: custom.props [hunk 7]
```diff
@@ -4,7 +4,7 @@
     <VersionMajor>1</VersionMajor>
     <VersionMinor>2</VersionMinor>
     <!-- The nuget package version should be incremented when we produce QFEs -->
-    <NuGetPackVersion>1.2.0-beta1</NuGetPackVersion>
+    <NuGetPackVersion>1.2.1</NuGetPackVersion>
```
Comment (shalinijoshi19) [inline custom.props]: Wy do we need to increment the AC package version? I thought this was a new package we were spinning up?

File: source/android/README.md [hunk 8]
```diff
@@ -25,6 +25,12 @@ swig -c++ -java -package io.adaptivecards.objectmodel -outdir main/java/io/adapt
 2. Select folder "<github directory>\AdaptiveCards\source\android"
 3. Build->"Rebuild Project"
 
+##or on a Mac
```
Comment (shalinijoshi19) [inline source/android/README.md]: >on a Mac [](start = 5, length = 8)

This is the mac section right?

File: source/android/adaptivecards/src/AdaptiveCardObjectModel.i [hunk 9]
```diff
@@ -386,7 +386,6 @@ namespace Json {
 %template(BaseActionElementVector) std::vector<std::shared_ptr<AdaptiveCards::BaseActionElement> >;
 %template(DateTimePreparsedTokenVector) std::vector<std::shared_ptr<AdaptiveCards::DateTimePreparsedToken> >;
 %template(ToggleVisibilityTargetVector) std::vector<std::shared_ptr<AdaptiveCards::ToggleVisibilityTarget> >;
-%template(StringVector) std::vector<std::string>;
```
Comment (shalinijoshi19) [inline source/android/adaptivecards/src/AdaptiveCardObjectModel.i]: Question: Why the delete?

**Comment by shalinijoshi19** [comment]
> 		{

Do we not care about this sample payload or did we move it out? 

---
Refers to: source/community/reactnative/src/visualizer/payloads/payloads/ColumnSet.Empty.json:6 in 520ddb6. [](commit_id = 520ddb64e77eeb83dabb06195133b07dfd691570, deletion_comment = True)

File: source/xamarin/AdaptiveCards.Rendering.Xamarin.iOS.Sample/Main.cs [hunk 10]
```diff
@@ -0,0 +1,15 @@
...
+namespace XamarinAC
+{
+    public class Application
+    {
+        // This is the main entry point of the application.
+        static void Main(string[] args)
+        {
+            // if you want to use a different Application Delegate class from "AppDelegate"
+            // you can specify it here.
+            UIApplication.Main(args, null, "AppDelegate");
```
Comment (shalinijoshi19) [inline source/xamarin/AdaptiveCards.Rendering.Xamarin.iOS.Sample/Main.cs]: Nit: Needs a copyright header added

File: source/xamarin/AdaptiveCards.Rendering.Xamarin.iOS.Sample/AppDelegate.cs [hunk 11]
```diff
@@ -0,0 +1,261 @@
...
+{
+    // The UIApplicationDelegate for the application. This class is responsible for launching the
+    // User Interface of the application, as well as listening (and optionally responding) to application events from iOS.
+    [Register("AppDelegate")]
+    public class AppDelegate : UIApplicationDelegate
+    {
+        // class-level declarations
+
+        public override UIWindow Window
+        {
```
Comment (shalinijoshi19) [inline source/xamarin/AdaptiveCards.Rendering.Xamarin.iOS.Sample/AppDelegate.cs]: also needs copyright header

**Comment by paulcam206** [comment]
closing out old PRs - feel free to revive this PR and resubmit for review

**Pull request closed without merging**